pwa-devdocs/src/tutorials/intercept-a-target/add-new-environment-variable/index.md

```diff
@@ -0,0 +1,259 @@
+
+```sh
+mkdir -p src/targets && touch src/targets/intercept.js
+```
+
+Set the value for `pwa-studio.targets.intercept` in your project's `package.json` file to tell the build process where to find the intercept file.
+
+```diff
+  {
+    "name": "PlaceholderImage",
+    "version": "1.0.0",
+    "license": "MIT",
+    "main": "src/components/PlaceholderImage/placeholderImage.js",
+    "peerDependencies": {
+      "@magento/pwa-buildpack": "^7.0.0",
+      "@magento/venia-ui": "^5.0.0",
+      "react": "^16.14.0"
+-   }
++   },
++   "pwa-studio": {
++     "targets": {
++       "intercept": "src/targets/intercept"
++     }
++   }
+  }
+```
+
+## Define the intercept file
+
+The intercept file is where you tap into PWA Studio's extensibility framework and add your modifications.
+
+In your intercept file, add the following content:
+
+```js
+module.exports = (targets) => {
+  const buildpackTargets = targets.of("@magento/pwa-buildpack");
+
+  buildpackTargets.envVarDefinitions.tap((defs) => {
+    defs.sections.push({
+      name: "PlaceholderImage settings",
+      variables: [
+        {
+          name: "IMAGE_PLACEHOLDER_SERVICE_URL",
+          type: "str",
+          desc: "Service URL for image placeholders",
+        },
+      ],
+    });
+  });
+
+  buildpackTargets.specialFeatures.tap((featuresByModule) => {
+    featuresByModule["PlaceholderImage"] = {
+      esModules: true,
+    };
+  });
+};
+```
+
+When this file runs, it taps into the `envVarDefinitions` target from the available targets in `@magento/pwa-buildpack` and passes in an intercept function.
+The intercept function appends a new definition to the [core environment variable definitions][], which allows frontend code access to the `IMAGE_PLACEHOLDER_SERVICE_URL` environment variable.
+
+## Test on a local instance
+
+Install this package in a local storefront project to use the PlaceholderImage component.
+
+```sh
+yarn add --dev link:/path/to/your/project
+```
+
+This adds a `devDependencies` entry to your storefront project's `package.json` that looks like the following:
+
+```diff
+    "@storybook/react": "~5.2.6",
++   "PlaceholderImage": "link:/path/to/your/project",
+    "apollo-cache-persist": "~0.1.1",
+```
+
+### Create environment variable entry
+
+In your project's `.env` file, create an entry for `IMAGE_PLACEHOLDER_SERVICE_URL`.
+
+```text
+IMAGE_PLACEHOLDER_SERVICE_URL=http://www.loremflickr.com/${w}/${h}
+```
+
+The PlaceholderImage component uses this value to compute the source when it renders the image.
+
+### Create a demo page
+
+Create a demo page component to render the ImagePlaceholder component.
+
+```sh
+mkdir -p src/components/PlaceholderImageDemo && touch src/component/PlaceholderImageDemo/placeholderImageDemo.js
+```
+
+Inside `placeholderImageDemo.js`, add the following content:
+
+```jsx
+import React from 'react'
+import PlaceholderImage from 'PlaceholderImage'
+
+const PlaceholderImageDemo = props => {
```

Comment:
Alright, I have to ask. The code style here matches the rules in the monorepo, but the examples above don't appear to have been run through the same ruleset. Not sure if you bounce back and forth depending if it's JSX or vanilla, just curious what the expectation is for consistency.

Reply:
I didn't really consider consistency or code style when I created this example. What needs to be changed to achieve consistency?

Reply:
Quotes and the types of things eslint removes that are unneeded jump out. eg. `const PlaceholderImageDemo = (props) => {`, parens aren't needed there. I have no preference if all code samples follow the monorepo rules, but do think they should all be formatted the same, whatever style that may be.

Reply:
got it. I probably copied from CodeSandbox and a local copy which would explain some of the inconsistencies. CodeSandbox formats every time I save, and I never ran any sort of linter on my local files.